File: week3.2/sum_of_digit.py
def sum_of_digit(num):
    sum = 0
    while num> 0:
        sum += num % 10
        num = num // 10
    return sum
# print(sum_of_digit(12345))

class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num >=10:
            sum = 0
            while num >0:
                sum += num % 10
                num = num // 10
            num = sum
        return num

s = Solution()
print(s.addDigits(12345))

File: week3.2/sum_of_digit.js
function sumOfDigit(num){
    // approch -1
    // =============================
    // const numStr = num.toString()
    // let sum = 0
    // for(let ele of numStr){
    //     sum += Number(ele)
    // }
    // return sum
    // ===========================
    // approch -2
    // ===========================
    // let sum = 0
    // while(num>0){
    //     sum += num%10
    //     num = Math.floor(num/10)
    // }
    // return sum
    // ===========================
    // approch -3
    // ===========================
    
    while(num>=10){
        let sum = 0
        while(num>0){
            sum += num%10
            num = Math.floor(num/10)
        }
        num = sum
        
    }
    
    return num
}
console.log(sumOfDigit(38))